packages/pigeon/example/app/linux/messages.g.h

```diff
@@ -0,0 +1,99 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+// Autogenerated from Pigeon (v11.0.1), do not edit directly.
+// See also: https://pub.dev/packages/pigeon
+
+#ifndef PIGEON_MESSAGES_G_H_
+#define PIGEON_MESSAGES_G_H_
+
+#include <flutter_linux/flutter_linux.h>
+
+G_BEGIN_DECLS
+
+typedef enum { MY_CODE_ONE = 0, MY_CODE_TWO = 1 } MyCode;
+
+G_DECLARE_FINAL_TYPE(MyMessageData, my_message_data, MY, MESSAGE_DATA, GObject)
+
+MyMessageData* my_message_data_new(MyCode code, FlValue* data);
+
+MyMessageData* my_message_data_new_full(const gchar* name,
+                                        const gchar* description, MyCode code,
+                                        FlValue* data);
+
+const gchar* my_message_data_get_name(MyMessageData* object);
+
+const gchar* my_message_data_get_description(MyMessageData* object);
+
+MyCode my_message_data_get_code(MyMessageData* object);
+
+FlValue* my_message_data_get_data(MyMessageData* object);
+
+G_DECLARE_FINAL_TYPE(MyExampleHostApi, my_example_host_api, MY,
+                     EXAMPLE_HOST_API, GObject)
+
+typedef struct {
+  void (*get_host_language)(
+      MyExampleHostApi* object,
+      FlBasicMessageChannelResponseHandle* response_handle, gpointer user_data);
+  void (*add)(MyExampleHostApi* object,
+              FlBasicMessageChannelResponseHandle* response_handle, int64_t a,
+              int64_t b, gpointer user_data);
+  void (*send_message)(MyExampleHostApi* object,
+                       FlBasicMessageChannelResponseHandle* response_handle,
+                       MyMessageData* message, gpointer user_data);
+} MyExampleHostApiVTable;
```

Comment:
Yes the plugin author implements the vtable (this pattern is copied from other GLib code, e.g. GDBusConnection).

Agreed, making separate handles for each method seems like a good idea.

Reply:
Added per method handles.

Reply:
Why per-method rather than per-type handles? E.g., if I have ten methods that all return bools and another one that returns an int, it seems like we only need two handles, not eleven.

Reply:
It's easier to generate them per-method rather than having to check if two are the same. I think the API will be clearer by not sharing handles and there should be no significant overhead (the compiler would probably optimize them together anyway).